releasenotes/notes/singletons-83782de8bd062cbc.yaml

```
@@ -0,0 +1,122 @@
+---
+features:
+  - |
+    Introduced a new class :class:`~.SingletonGate` which is a subclass of
+    :class:`~.Gate` that uses a single instance for all objects of that type.
+    The intent behind this class is to minimize the memory and construction
+    overhead of using multiple gates in a circuit with the tradeoff of having
+    global shared state. For this reason this class is only applicable to
+    gates that do not have any unique and/or mutable state stored in an instance.
+    For example, the best example of this is :class:`~XGate` doesn't contain
```

```suggestion
    For example, the best example of this is :class:`.XGate` doesn't contain
```
(I don't think the `~XGate` lookup would succeed.)